res/res_rtp_asterisk.c

```diff
@@ -9191,6 +9209,14 @@ static int rtp_red_buffer(struct ast_rtp_instance *instance, struct ast_frame *f
 
 	if (frame->datalen > 0) {
 		if (red->t140.datalen > 0) {
+			/*
+			* Avoid merging command and regular T.140 text packets
+			* When realtime text packets are to be sent, the text is accumulated in a
+			* buffer and sent regularly by a timer.  It can happen that commands such
+			* as a backspace, CR, or LF get merged with regular text. This breaks some
+			* UAs. In recent tests it could not be reproduced, but its probably better
+			* to keep it.
+			*/
```

Comment:
This comment seems irrelevant to this PR? Doesn't really matter to me just wanted to point out that something unintended may have snuck in.

Reply:
This comment was added to explain the logic in the block below. We observed that its not executed with our two different test clients, but it was added for a reason and we did not wanted to remove it. Maybe it can be moved down below the variable assignment, if this is more clear.